### DISCUSSION TITLE
How do I communicate with .socket.sock?

### DISCUSSION THREAD

**Discussion by pb-kh** [hyprwm/Hyprland#3616] [Q&A] [answered]
Trying to get devices info through this socket, sending the following message:
```c
char msg[] = "j/devices";
write(sockets->sock1, msg, sizeof(msg));
```

but the app stops from the SIGPIPE signal... What am i doing wrong?

P.S. The socket is correct, value `4`

**Comment by vaxerski** [comment, marked as answer]
you can read the hyprctl source. Generally you probably shouldn't rawdog socket1.

you write your stuff, you `read()` instantly for a reply, and once you get one you `close()`

**Reply by pb-kh** [reply]
thank you! hyprctl source helped a lot. it turns out, i had to use `SUN_LEN` macro instead of plain sizeof...
